polars/polars-core/src/series/mod.rs

```diff
@@ -511,11 +512,18 @@ impl Series {
     }
 
     /// Get the sum of the Series as a new Series of length 1.
+    /// Returns a Series with a single zeroed entry if self is an empty numeric series.
     ///
     /// If the [`DataType`] is one of `{Int8, UInt8, Int16, UInt16}` the `Series` is
     /// first cast to `Int64` to prevent overflow issues.
     pub fn sum_as_series(&self) -> Series {
         use DataType::*;
+        if self.is_empty() && self.dtype().is_numeric() {
+            return Series::new("", [0])
```

Comment:
Almost there. `Int8 | UInt8 | Int16 | UInt16` should be `Int64`.

Reply:
Since we recursively call into `sum_as_series`, won't they be converted to Int64 on the second call? Or should I implement this logic to save on an unnecessary call?

Reply:
Oh, right. Missed that one. 

Nope. This is good!

Reply:
Awesome! Thank you for your feedbacks on this PR 😸